src/main/java/com/synopsys/arc/jenkinsci/plugins/jobrestrictions/restrictions/job/AbstractUserCauseRestriction.java

```diff
@@ -114,12 +115,31 @@ public final boolean isCheckUpstreamProjects() {
 
     @Override
     public boolean canTake(Queue.BuildableItem item) {
-        final List<Cause> causes = new ArrayList<Cause>();
-        for (Action action : item.getActions()) {
-            if (action instanceof CauseAction) {
-                CauseAction causeAction = (CauseAction) action;
-                causes.addAll(causeAction.getCauses());
-            } 
+        final List<Cause> causes;
+        
+        // The enclosed BuildableItem has a Pipeline step task
+        if (item.task.getClass().equals(PlaceholderTask.class)) {
+            // This tasks's context is present, causes can be retrieved
+            if (((PlaceholderTask) item.task).run() != null) {
+                causes = ((PlaceholderTask) item.task).run().getCauses();
+            // This task's context has not loaded yet, mostly likely due to a Jenkins' restart
+            } else {
+                // Context loaded and is now present
+                if (((PlaceholderTask) item.task).runForDisplay() != null) {
+                    causes = ((PlaceholderTask) item.task).runForDisplay().getCauses();
+                // Context cannot be loaded
+                } else {
+                    causes = null;
```

Comment:
It will cause NPE in the `canTake(causes)`. Empty list + logging should be used instead IMHO